lib/Analysis/LazyRelinAnalysis/LazyRelinAnalysis.cpp

```diff
@@ -0,0 +1,354 @@
+#include "lib/Analysis/LazyRelinAnalysis/LazyRelinAnalysis.h"
+
+#include "lib/Dialect/BGV/IR/BGVDialect.h"
+#include "lib/Dialect/BGV/IR/BGVOps.h"
+#include "llvm/include/llvm/ADT/DenseMap.h"             // from @llvm-project
+#include "llvm/include/llvm/ADT/TypeSwitch.h"           // from @llvm-project
+#include "llvm/include/llvm/Support/Debug.h"            // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/Operation.h"             // from @llvm-project
+#include "mlir/include/mlir/IR/Value.h"                 // from @llvm-project
+#include "ortools/linear_solver/linear_solver.h"  // from @com_google_ortools
+
+// For now, fix the key basis degree bound to 3. Could make it a pass flag
+// later, or introduce costs for high-degree operations and let it be
+// unconstrained.
+constexpr int MAX_KEY_BASIS_DEGEE = 3;
```

Comment:
Does a key basis degree of three mean three polynomials per ctxt and keybasis (1, s, s^2) or does it mean the highest degree in the keybasis is three (1,s,s^2,s^3)? I'm guessing the latter, but an example in the comment here probably wouldn't hurt

nit: typo in name
```suggestion
constexpr int MAX_KEY_BASIS_DEGREE = 3;
```

Reply:
Done